uaa/src/main/webapp/WEB-INF/spring/saml-providers.xml

```diff
@@ -78,6 +78,14 @@
             <constructor-arg value="${login.saml.entityIDAlias:${login.entityID:unit-test-sp}}"/>
         </bean>
 
+        <bean id="globalSamlSignatureAlgorithm" class="org.cloudfoundry.identity.uaa.zone.SamlConfig$SignatureAlgorithm" factory-method="valueOf">
+            <constructor-arg value="${login.saml.signatureAlgorithm.global:SHA256}"/>
+        </bean>
+
+        <bean id="defaultUaaSamlSignatureAlgorithm" class="org.cloudfoundry.identity.uaa.zone.SamlConfig$SignatureAlgorithm" factory-method="valueOf">
+            <constructor-arg value="${login.saml.signatureAlgorithm:#{globalSamlSignatureAlgorithm.name()}}"/>
```

Comment:
is the yml path for this supposed to be the same as `globalSamlSignatureAlgorithm`? I though UAA zone is supposed to have it's own property

Reply:
UAA Zone will have its own property `login.saml.signatureAlgorithm`. But, we would need a fallback if that value isn't configured, which is the `globalSamlSignatureAlgorithm`. The same applies to `ClientSecretPolicy`